crates/sui-core/src/consensus_handler.rs

```diff
@@ -151,7 +169,7 @@ impl<T: ObjectStore + Send + Sync, C: CheckpointServiceNotify + Send + Sync> Exe
 
         let last_committed_round = self.last_consensus_stats.index.last_committed_round;
 
-        let round = consensus_output.sub_dag.leader_round();
+        let round = consensus_output.leader_round();
 
         assert!(round >= last_committed_round);
         if last_committed_round == round {
```

Comment:
I believe we need to think the multi leader commit case the Mysticeti supports , as _I assume_ we'll have cases where committed sub dags will have leaders of the same round - we should allow them to process in that case. Unless the whole vector of sub dags are pre-sorted and committed as one the consensus output - assuming no partial round commits are allowed - so then everything could be hidden as implementation detail of the mysticeti consensus output. Haven't thought that much. Probably not a concern for this PR, but leaving as observation. CC @andll

Reply:
Yes, we'll need to address - eventually we need to just move away from using `round` at all(e.g. remove `leader_round` function from the trait) and rely fully on the commit index. But for now we should just remember to disable multi-leader in mysticeti when integrating, so that it does not break things